hathor/manager.py

```diff
@@ -993,11 +1001,11 @@ def step_validations(self, txs: Iterable[BaseTransaction]) -> None:
                 # TODO
                 raise
             else:
-                self.tx_storage.save_transaction(tx, only_metadata=True, add_to_indexes=True)
+                self.tx_storage.save_transaction(tx, only_metadata=True)
+                self.tx_storage.add_to_indexes(tx)
                 self.consensus_algorithm.update(tx)
-                # save and process its dependencies even if it became invalid
-                # because invalidation state also has to propagate to children
-                self.tx_storage.indexes.deps.remove_ready_for_validation(tx.hash)
+                self.tx_storage.indexes.update(tx)
+                self.tx_storage.indexes.mempool_tips.update(tx)
```

Comment:
Maybe it will work now, it didn't work when I split it into pre/pos. But I'd rather not experiment with this at this point. I'll add a comment that it should be moved to `update`.